sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -627,83 +627,67 @@ class GpuHashAggregateIterator(
     val aggModeCudfAggregates =
       AggregateUtils.computeAggModeCudfAggregates(aggregateExpressions, aggBufferAttributes)
 
+    // boundInputReferences is used to pick out of the input batch the appropriate columns
+    // for aggregation.
     //
-    // expressions to pick input to the aggregate, and finalize the output to the result projection.
-    //
-    // Pick update distinct attributes or input projections for Partial
-    val (distinctAggExpressions, nonDistinctAggExpressions) = aggregateExpressions.partition(
-      _.isDistinct)
-    val updateExpressionsDistinct =
-      distinctAggExpressions.flatMap(
-        _.aggregateFunction.updateExpressions)
-    val updateAttributesDistinct =
-      distinctAggExpressions.flatMap(_.aggregateFunction.aggBufferAttributes)
-    val inputProjectionsDistinct =
-      distinctAggExpressions.flatMap(_.aggregateFunction.inputProjection)
-
-    // Pick merge non-distinct for PartialMerge
-    val mergeExpressionsNonDistinct =
-      nonDistinctAggExpressions
+    // - PartialMerge with Partial mode: we use the inputProjections or distinct update expressions
+    //   for Partial and non distinct merge expressions for PartialMerge.
+    // - Final or PartialMerge-only mode: we pick the columns in the order as handed to us.
+    // - Partial or Complete mode: we use the inputProjections or distinct update expressions.
+    val boundInputReferences =
+    if (modeInfo.hasPartialMerge && modeInfo.uniqueModes.contains(Partial)) {
```

Comment:
```suggestion
    if (modeInfo.hasPartialMerge && modeInfo.hasPartialMode) {
```

Reply:
bonus points => `hasPartialMerge` => `hasPartialMergeMode`

Reply:
We used `modeInfo.uniqueModes.contains(Partial)` instead of `modeInfo.hasPartialMode` here,  because hasPartialMode is set as `hasPartialMerge || uniqueModes.contains(Partial)`. I adjusted `AggregateModeInfo` to make its members more consistent with their names.